MAUI/DataForm/Editors.md

```diff
@@ -0,0 +1,1187 @@
+
+{% endhighlight %}
+{% endtabs %}
+
+`Image`
+
+## RadioGroup editor
+
+In the [RadioGroup](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/radiobutton?view=net-maui-7.0) editor, the [RadioButton](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/radiobutton?view=net-maui-7.0) control is loaded.
+
+The `items` for [RadioButton](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/radiobutton?view=net-maui-7.0) is generated for `enum` and `List` data type properties. In order to add [RadioButton](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/radiobutton?view=net-maui-7.0) editor in the DataForm, you need to register editor as DataFormEditorType.RadioGroup for the required property by using the `RegisterEditor` method.
+
+**Support for enum data type**
+
+For `enum` data type property, [RadioButon](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/radiobutton?view=net-maui-7.0) `items` will be added based on specified property enum values.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.RegisterEditor("Phone", DataFormEditorType.RadioGroup);
+
+    public Numbers Phone  { get; set; }
+
+    public enum Numbers
+    {
+	    Home,
+	    Work,
+	    Other
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+**Support for List data type**
+
+For `List` data type property, you have to set the `ItemsSource` by using the `IDataFormSourceProvider`, based on that [RadioButton](https://learn.microsoft.com/en-us/dotnet/maui/user-interface/controls/radiobutton?view=net-maui-7.0) `items` will be added.
+
+{% tabs %}
+{% highlight C# %}
+
+    dataForm.RegisterEditor("Phone", DataFormEditorType.RadioGroup);
+    dataForm.ItemsSourceProvider = new DataFormItemsSourceProvider();
+
+    public string Phone  { get; set; }
+
+    public class DataFormItemsSourceProvider : IDataFormSourceProvider
+    {
+         public object GetSource(string sourceName)
+        {
+		    if(sourceName == "Phone")
+		    {
+                List<string> list = new List<string>() {"Home", "Work", "Other"};
+                return list;
+		    }
+		    return new List<string>();
+        }
+    }
+
+{% endhighlight %}
+{% endtabs %}
+
+`Image`
+	
+## Custom editor
+
+The custom editor can be added to DataForm by inheriting the `IDataFormEditor` class for business models. You can create custom editor using implement all method in IDataFormEditor. 
+
+To add custom editor in DataForm, register the editor with custom registered type for the required property using `RegisterEditor` method. You can also customize editor settings by using available methods in `IDataFormEditor`. 
+
+#### Creating custom editor using IDataFormEditor 
+
+Views such as labels, buttons, and sliders can be loaded to custom editor. Here, entry is loaded as custom editor for `PhoneNumer` property. 
+
+#### Below methioned example code for Numeric editor which accepts only numbers 
+
+{% tabs %}
+{% highlight C# %}
+
```

Comment:
public class NumericTextEditor : IDataFormEditor
{ 
        private SfDataForm dataForm;

        private DataFormCustomItem? dataFormCustomItem;

        public NumericEditor(SfDataForm dataForm)
        {
            this.dataForm = dataForm;
        }

        public View CreateEditorView(DataFormItem dataFormItem) 
        { 
            Entry inputView = new Entry();
            inputView.Keyboard = Keyboard.Numeric;
            inputView.Placeholder = dataFormItem.PlaceholderText;
            DataFormTextStyle textStyle = dataForm.EditorTextStyle;
            inputView.TextColor = textStyle.TextColor;
            inputView.FontSize = textStyle.FontSize;
            inputView.FontFamily = textStyle.FontFamily;
            inputView.FontAttributes = textStyle.FontAttributes;
            inputView.TextChanged += this.OnViewTextChanged;
            this.dataFormCustomItem = (DataFormCustomItem)dataFormItem;

this.dataFormCustomItem.EditorValue = string.Empty;
            return inputView; 
        } 

        public void CommitValue(DataFormItem dataFormItem, View view)
        {
            if (view is InputView numericText)
            {
                double numericValue;
                double.TryParse(numericText.Text, out numericValue);
                 dataFormItem.SetValue(numericValue);
            }
        }

        private void ValidateValue(DataFormItem dataFormItem)
        {
            dataForm.Validate(new List<string>() { dataFormItem.FieldName });
        }

        private void OnViewTextChanged(object? sender, TextChangedEventArgs e)
        {
            if (sender is not InputView numericEntry || dataFormCustomItem == null)
            {
                return;
            }

            string? numericText = Regex.Replace(numericEntry.Text, "[^0-9]+", string.Empty);
            if (numericText != numericEntry.Text)
            {
                numericEntry.Text = numericText;
                return;
            }

            dataFormCustomItem.EditorValue = numericText;
            this.ValidateValue(dataFormCustomItem);
            this.CommitValue(dataFormCustomItem, numericEntry);
        }

        public void UpdateReadyOnly(DataFormItem dataFormItem)
        {
        }
    }

Reply:
code updated